### PR TITLE
Finished Weather project

### PR DESCRIPTION
**Pull request by webmarkyn**
Microverse Weather project.
Now you are able to:
-Choose a city
-Choose a metric system
-Get an emoji representing Weather state
Built with:
HTML,CSS,JS,Webpack, Node.js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "weather-app",
+  "version": "1.0.0",
+  "description": "",
+  "main": "main.js",
+  "scripts": {
```
Comment (adriaanbd) [inline package.json:6]: Please add the other run commands, as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/scripts_requirement.png) [`REQUIRED`]:

```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  }
```

File: src/weather.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import weatherColor from './colors';
+import {
+  loader, overlay, mainInput, weatherInfo,
+} from './dom';
+import { openPopup, closePopup } from './events';
+
+async function getWeather(city, units) {
+  const url = `https://cors-anywhere.herokuapp.com/api.openweathermap.org/data/2.5/weather?q=${city}&units=${units}&appid=1a7d76daf845e6d9571a1071f31062f1`;
+  return fetch(url).then(data => data.json());
```
Comment (adriaanbd) [inline src/weather.js:9]: If you're using async, then you should use `await`. Any particular reason you chose `async` + `.then`? Requirements state that either `promise` or `async/await` should be used [`REQUIRED`], as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/promises_requirement.png).

File: src/weather.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  weatherColor[data.weather[0].main].color
+}">
+        ${weatherColor[data.weather[0].main].icon}
+      </div>
+      <div class="weather-desc">${data.weather[0].main}</div>
+      <div class="weather-temp">${data.main.temp}°</div>
+    `;
+}
+
+export default function loadWeather(city, units) {
```
Comment (adriaanbd) [inline src/weather.js]: Please use ES6 syntax, which implies arrow functions, as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) [`REQUIRED`]

File: src/weather.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  loader, overlay, mainInput, weatherInfo,
+} from './dom';
+import { openPopup, closePopup } from './events';
+
+async function getWeather(city, units) {
+  const url = `https://cors-anywhere.herokuapp.com/api.openweathermap.org/data/2.5/weather?q=${city}&units=${units}&appid=1a7d76daf845e6d9571a1071f31062f1`;
+  return fetch(url).then(data => data.json());
+}
+
+function renderWeather(data) {
```
Comment (adriaanbd) [inline src/weather.js]: Please use ES6 syntax, which implies arrow functions, as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) [`REQUIRED`]

File: src/weather.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import weatherColor from './colors';
+import {
+  loader, overlay, mainInput, weatherInfo,
+} from './dom';
+import { openPopup, closePopup } from './events';
+
+async function getWeather(city, units) {
```
Comment (adriaanbd) [inline src/weather.js]: Please use ES6 syntax, which implies arrow functions, as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) [`REQUIRED`]

File: src/events.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+function openPopup(popup) {
+  popup.style.visibility = 'visible';
+  popup.style.opacity = '1';
+}
+
+function closePopup(popup) {
```
Comment (adriaanbd) [inline src/events.js]: Please use ES6 syntax, which implies arrow functions, as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) [`REQUIRED`]

File: main.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+            user.email = data["email"] if user.email.blank?    
+        end  
+    end 
+end 
+def self.from_omniauth(auth)  
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|    
+        user.email = auth.info.email    
+        user.password = Devise.friendly_token[0,20]  
+    end 
+end
```
Comment (adriaanbd) [inline main.rb]: We are in the JavaScript section, therefore please find the JS way of doing this. Personally, it doesn't make sense to use Ruby in a vanilla JS section project. You'll have a chance to work with both in the final graduation Capstone project.

**Review by adriaanbd — CHANGES_REQUESTED** [review]
Please follow the Git Flow convention of setting up the default branch of the repo to `develop`.

See my other comments below.

Status: `Changes Requested`
Reviewer: `Adriaan Beiertz`

File: dist/styles.css [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+#overlay {
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    background: #fff;
+    transition: 0.1s;
+    /* display: none; */
```
Comment (bellom) [inline dist/styles.css:15]: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))


**To do this with all your commented code.**

**Review by bellom — APPROVED** [review]
## Hi @webmarkyn  :clap: 

**Great work making all the necessary corrections from the previous review.**

**_I trust you can attend to my comments below before merging._**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**